src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs

```diff
@@ -696,5 +695,23 @@ private bool IsSuspended()
         {
             return _initializing || DesignMode;
         }
+
+        private class NormalizedFilterCollection : Collection<string>
+        {
+            protected override void InsertItem(int index, string item)
+            {
+                base.InsertItem(index, string.IsNullOrEmpty(item) || item == "*.*" ? "*" : item);
+            }
+
+            protected override void SetItem(int index, string item)
+            {
+                base.SetItem(index, string.IsNullOrEmpty(item) || item == "*.*" ? "*" : item);
+            }
+
+            public override string ToString()
```

Comment:
Don't override `ToString()`. There is no end-user value in a custom `ToString()`.